File: reverse.whois/ruby/reverse_api_example_post.rb
require 'json'
require 'yaml'		# only needed to print the returned result in a very pretty way
require 'uri'
require 'openssl'
require "net/https"

########################
# Fill in your details #
########################
username = "YOUR_USERNAME"
password = "YOUR_PASSWORD"

#######################
# Use a JSON resource #
#######################
format = "JSON"
url = 'https://www.whoisxmlapi.com/reverse-whois-api/search.php'

content = {
    "terms" => [
        {
        section: "Registrant",
        attribute: "Name",
        matchType: "BeginsWith",
        value: "Mark",
        exclude:"true"
        },
        {
        section: "Technical",
        attribute: "Country",
        matchType: "Anywhere",
        value: "US",
        exclude:"true"
        }
    ],
    recordsCounter: "false",
    username: username,
    password: password ,
    output_format: format,
}

uri = URI.parse(url)
http = Net::HTTP.new(uri.host, uri.port)

# connect using ssl
http.use_ssl = true
http.verify_mode = OpenSSL::SSL::VERIFY_NONE
request = Net::HTTP::Post.new(uri.request_uri)

# set headers
request.add_field('Content-Type', 'application/json')
request.add_field("Accept", "application/json")
request.body = content.to_json

# get response
response = http.request(request)

# print pretty parsed json
puts JSON.parse(response.body).to_yaml

File: reverse.whois/perl/reverse_api_example.post.pl
#!/usr/bin/perl

use LWP::Simple;                # From CPAN
use JSON qw( decode_json );     # From CPAN
use Data::Dumper;               # Perl core module
use strict;                     # Good practice
use warnings;                   # Good practice

use LWP::UserAgent;
use HTTP::Request::Common qw{ POST };
use CGI;


########################
# Fill in your details #
########################
my $user_name = "";
my $password = "";

my $url = 'https://www.whoisxmlapi.com/reverse-whois-api/search.php';

#######################
# Use a JSON resource #
#######################

my $responseJson = JSON->new->decode(reverseWhoisApiSearch("json"));
print "JSON\n---\n";
print JSON->new->pretty->encode($responseJson);

#######################
#     getting Data    #
#######################

sub reverseWhoisApiSearch {
    my $content = ' {
        "terms":[
            {
                "section":"Admin",
                "attribute":"organization",
                "value":"Wikimedia",
                "matchType":"anywhere",
                "exclude":"false"
            },
            {
                "section": "technical",
                "attribute": "city",
                "value": "San Francisco",
                "matchType":"anywhere",
                "exclude":"false"
            }
        ],

        "recordsCounter":false,
        "outputFormat":"json",
        "username":"'.$user_name.'",
        "password":"'.$password.'",
        "rows":100
    }';

    my $ua = LWP::UserAgent->new(ssl_opts => { verify_hostname => 0 });
    my $req = HTTP::Request->new("POST", $url);

    $req->header('Content-Type' => 'application/json');
    $req->header("Accept", "application/json");
    $req->content($content);

    my $response = $ua->request($req);
    print $response->content;
    return $response->content;

}
